### ISSUE TITLE
syntax highlight incorrect when use backtick inside a function and use destrcuting argument in the function

### ISSUE THREAD

**Issue by maple-leaf** [HerringtonDarkholme/yats.vim#165]
My neovim config:

```viml
if empty($XDG_DATA_HOME)
    let $XDG_DATA_HOME = $HOME . '/.config'
endif

let mapleader = "\<Space>"

function! s:checkPlug()
	let vimplug_exists=expand('~/.config/nvim/autoload/plug.vim')

	if !filereadable(vimplug_exists)
	  if !executable("curl")
	    echoerr "You have to install curl or first install vim-plug yourself!"
	    execute "q!"
	  endif
	  echo "Installing Vim-Plug..."
	  echo ""
	  silent !\curl -fLo ~/.config/nvim/autoload/plug.vim --create-dirs https://raw.githubusercontent.com/junegunn/vim-plug/master/plug.vim
	  let g:not_finish_vimplug = "yes"

	  autocmd VimEnter * PlugInstall
	endif
endfunction

call s:checkPlug()

" Required:
call plug#begin(expand('~/.config/nvim/plugged'))

Plug 'HerringtonDarkholme/yats.vim'

call plug#end()
```
pls check last few lines in the screenshot:
![image](https://user-images.githubusercontent.com/3980995/75851658-3db08180-5e25-11ea-8067-2190c43b96fa.png)

**Comment by HerringtonDarkholme** [comment]
MInimal production 

```ts
[].map(({num}, index) => {
    const key = `${num}`
})

```

**Comment by Schniz** [comment]
Also doesn’t work:
```typescript
<styled jsx>{`
  .something {
  }
`}</styled>
```

**Comment by HerringtonDarkholme** [comment]
fixed in **99d3554fb6f958256e61bbcfcf79a5ac5f6a9c36**
